lib/optimizely/helpers/validator.rb

```diff
@@ -95,6 +95,27 @@ def string_numeric?(str)
       rescue
         false
       end
+
+      def inputs_valid?(variables, logger = NoOpLogger.new, level = Logger::ERROR)
+        # Determines if values of variables in given array are non empty string.
+        #
+        # variables - array values to validate.
+        #
+        # logger - logger.
+        #
+        # Returns boolean True if all of the values are valid, False otherwise.
+
+        return false unless variables.respond_to?(:each) && !variables.empty?
+        is_valid = true
+        variables.each do |key, value|
+          next if value.is_a?(String) && !value.empty?
+          is_valid = false
```

Comment:
I don't think this is correct and your tests seem to confirm this.  If the last argument is invalid, everything is fine,  but, if your middle argument is invalid, it would not catch it.  Shouldn't it be is_valid ||= false?  Or return when you get the first is_valid = false.  Also, can you add tests that actually call the activate with a integer user id or a empty experiment key (first argument)?